examples/aws/test-scale.py

```diff
@@ -151,6 +151,14 @@ def greater_than_zero(node_count):
             else:
                 node_count += args.increment
 
+            # TODO Feature Request 
```

Comment:
Ah, so you are saying, if we (for example) enter the loop at node_count == 5, and that is also the max size, when we increment it's going to go up to 6, but actually we don't want to go above 5 so we should not run cli.scale to begin with?

For the case of incrementing by 1 (what these scripts were originally written for) the `keep_going` function is checking only that it's less than or equal to the max, so perhaps we have an off by one error there and just want to check that it's less than the max (and then this allows us to increment by 1 without going over).

For the case when the increment isn't equal to one (e.g., if we start at 5 and are incrementing by 5, and would just go way over) I think what we probably want to do is tweak the functions to accept the increment as an argument, and then check if it's still below some max. I think we might also want to account for cases for when we are slightly under (but still want to go up to the max). E.g., if our max is 10 and we are going up by 3 and we hit 9, we want that last check to say "Oh, you can't go up by 3, but you can go up by 1"

Am I hitting the right vein or just totally off to the issue? As for the design - it's up to you - right now we are returning a boolean, but you could change it to be the case that we return the size to scale by, and then a return of 0 (that would still be able to use `while keep_going(node_count, args.increment):`. And actually, since the increment is set in stone we could probably just include it in the function without making it an arg.

Let me know your thoughts!

Reply:
Yes, you're spot on. Yeah, if we started the script say max_node_count is 5, and we're incrementing one, then our current script allow us to execute one more scaling when we are at node_count=5, and this returns in error because the stack max is 5 and stack update fails. +1 for pointing out the cases of suggesting scaling "Oh, you can't go up by 3, but you can go up by 1". Definitely, this will be important for N number of increments at a time. We can say, "You can't scale up by N, but the max you can is max_node_count - N". 

Also, another thing, we are by default starting with 1. so, if users wants e.g. increment from 1--10 (where 10 is max_node_count provided by args), technically they will be able to scale up to 9 times. if they wants to scale up 10 times by increment 1, they need to mention max_node_count as 11, right? 

I also like your idea of designing this feature. This way, we can reuse and implement without changing the architecture much.

Reply:
For AWS since the control plane starts at 0 we could also start at 0!

Do you want me to put in a PR for you to test or would you like to take a shot at implementing this idea?

Reply:
Yeah sure, please put it into PR, I'd love to test and let you know!

Reply:
okay here you go! This was written out of my ears, not tested, so let me know if there are issues! I opted for still maintaining two functions (that are heavily commented) so the logic is totally clear vs. making one "cleaner" function that can handle both cases (but would be harder for a human to parse) https://github.com/converged-computing/kubescaler/pull/9